ui/pages/reset_password.tsx

```diff
@@ -1,31 +1,106 @@
-import { Button } from "@canonical/react-components";
+import { SettingsFlow } from "@ory/client";
+import { Button, Form } from "@canonical/react-components";
 import type { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useEffect, useState, useCallback, FormEvent } from "react";
 import React from "react";
+import { handleFlowError } from "../util/handleFlowError";
+import { kratos } from "../api/kratos";
 import PageLayout from "../components/PageLayout";
 import Password from "../components/Password";
+import { UiNodeInputAttributes } from "@ory/client/api";
+import { AxiosError } from "axios";
 
 const ResetPassword: NextPage = () => {
   const [password, setPassword] = React.useState("");
   const [isPassValid, setPassValid] = React.useState(false);
+  const [flow, setFlow] = useState<SettingsFlow>();
+
+  // Get ?flow=... from the URL
+  const router = useRouter();
+  const { return_to: returnTo, flow: flowId } = router.query;
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return;
+    }
+
+    // If ?flow=... was in the URL, we fetch it
+    if (flowId) {
+      kratos
+        .getSettingsFlow({ id: String(flowId) })
+        .then((res) => setFlow(res.data))
+        .catch(handleFlowError(router, "settings", setFlow));
+      return;
+    }
+
+    // Otherwise we initialize it
+    kratos
+      .createBrowserSettingsFlow({
+        returnTo: returnTo ? String(returnTo) : undefined,
+      })
+      .then(({ data }) => {
+        if (data.request_url !== undefined) {
+          window.location.href = data.request_url;
+          return;
+        }
+        setFlow(data);
+      })
+      .catch(handleFlowError(router, "settings", setFlow))
+      .catch(async (err: AxiosError<string>) => {
+        if (err.response?.data.trim() === "Failed to create settings flow") {
+          setFlow(undefined);
+          await router.push("/login");
```

Comment:
```suggestion
          await router.push("./login");
```

When the UI implements the fix we discussed we can ditch the relative paths, but we need this for now

Reply:
replaced all the frontend paths with relative ones.